src/probnum/kernels/_exponentiated_quadratic.py

```diff
@@ -0,0 +1,68 @@
+"""Exponentiated quadratic kernel."""
+
+from typing import Optional
+
+import numpy as np
+import scipy.spatial.distance
+
+import probnum.utils as _utils
+from probnum.type import IntArgType, ScalarArgType
+
+from ._kernel import Kernel
+
+_InputType = np.ndarray
+
+
+class ExpQuad(Kernel[_InputType]):
+    """Exponentiated quadratic / RBF kernel.
+
+    Covariance function defined by :math:`k(x_0, x_1) = \\exp \\big(-\\frac{\\lVert
+    x_0 - x_1 \\rVert^2}{2l^2}\\big)`. This kernel is also known as the squared
+    exponential or radial basis function kernel.
+
+    Parameters
+    ----------
+    input_dim :
+        Input dimension of the kernel.
+    lengthscale
+        Lengthscale of the kernel. Describes the input scale on which the process
+        varies.
+
+    See Also
+    --------
+    RatQuad : Rational quadratic kernel.
+    Matern : Matern kernel.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from probnum.kernels import ExpQuad
+    >>> K = ExpQuad(input_dim=1, lengthscale=0.1)
+    >>> K(np.linspace(0, 1, 3)[:, None])
+    array([[1.00000000e+00, 3.72665317e-06, 1.92874985e-22],
+           [3.72665317e-06, 1.00000000e+00, 3.72665317e-06],
+           [1.92874985e-22, 3.72665317e-06, 1.00000000e+00]])
+    """
+
+    def __init__(self, input_dim: IntArgType, lengthscale: ScalarArgType = 1.0):
+        self.lengthscale = _utils.as_numpy_scalar(lengthscale)
+        super().__init__(input_dim=input_dim, output_dim=1)
+
+    def __call__(self, x0: _InputType, x1: Optional[_InputType] = None) -> np.ndarray:
+        # Check and reshape inputs
+        x0, x1, kernshape = self._check_and_reshape_inputs(x0, x1)
+
+        # Compute pairwise euclidean distances ||x0 - x1|| / l
+        if x1 is None:
```

Comment:
It used to be this way, but then I would have to return a flag whether the arguments were equal. Now `x1` is `None` if it's not passed. Every kernel that for example relies on `scipy.spatial.pdist` and `scipy.spatial.cdist` benefits from the knowledge whether `x0` and `x1` are the same or not.

Reply:
I am not sure we are talking about the same thing here... what I meant was that in line 160-162 of `self._check_and_reshape_inputs` one finds the following code
```python
        if x1 is None:
            x1 = x0
            equal_inputs = True ```
that would mean that x1 is not `None` after the call anymore?

Reply:
Okay cool thanks for taking care of this. Please dont hate me now, but I think the same updates now need to be done in the other kernel classes as well (at least in Polynomial, which is where I looked last?). Though if you like, you can also merge already and make an issue about this and fix it later. Whatever you reply, I'll give you the approval then :)

Reply:
I don't think so. `_check_and_reshape_inputs` now returns `None` for `x1` if only `x0` was passed. This can then be checked in the subclasses to potentially use a more efficient method (such as `pdist`) to compute the kernel matrix.